include/sentry.h

```diff
@@ -1364,6 +1392,8 @@ SENTRY_EXPERIMENTAL_API sentry_uuid_t sentry_transaction_finish(
  * Invoke `sentry_transaction_finish` to remove the Span set by this function as
  * well as its reference by passing in the same Transaction as the one passed
  * into this function.
+ *
+ * The Transaction passed in must have a read lock to be thread-safe.
```

Comment:
```suggestion
 * This mutates the transaction passed in, which is not thread-safe by itself.  To
 * mutate the transaction from multiple threads a lock must must be used to protect
 * mutations.
```

The "read lock" part is somewhat odd too, sorry if this feels like nitpicking.  The suggestion here is pretty verbose, can probably be made shorter too.